engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -278,4 +285,72 @@ public static synchronized void changeLoggersLevel(@Nonnull Map<String, String>
     }
     return ImmutableMap.copyOf(matchedClasses);
   }
+
+  /**
+   * Returns logger log file {@link LogFileInformation}
+   * @return logFileInformation
+   */
+  @Evolving
+  public static LogFileInformation getLogFileInformation()
+  {
+    if (shouldFetchLogFileInformation()) {
+      File logFile = new File(rootFileAppender.getFile());
+      LogFileInformation logFileInfo = new LogFileInformation(rootFileAppender.getFile(), logFile.length());
+      return logFileInfo;
+    }
+    return new LogFileInformation(null, 0);
+  }
+
+  private static FileAppender getRootFileAppender()
+  {
+    Enumeration e = LogManager.getRootLogger().getAllAppenders();
+    FileAppender fileAppender = null;
+    while (e.hasMoreElements()) {
+      Appender appender = (Appender)e.nextElement();
+      if (appender instanceof FileAppender) {
+        if (fileAppender == null) {
+          fileAppender = (FileAppender)appender;
+        } else {
+          //skip fetching log file information if we have multiple file Appenders
+          return null;
+        }
+      }
+    }
+    return fileAppender;
+  }
+
+  /*
+   * We should return log file information only if,
+   * we have single file Appender, the logging level of appender is set to level Error or above and immediateFlush is set to true.
+   * In future we should be able to enhance this feature to support multiple file appenders.
+   */
+  private static boolean shouldFetchLogFileInformation()
+  {
+    if (rootFileAppender != null && isErrorLevelEnable() && rootFileAppender.getImmediateFlush()) {
+      return true;
+    }
+    logger
```

Comment:
I did not understand your comment "In case this is done during static initialization, I would change log level from debug to info as it is one time logging."
Reason I am not doing it during static initialization is, the logging level can be changed anytime, even through UI. In case someone changes logging level which turns of logging on exception there is not point in fetching log file information.